tools/memap.py

```diff
@@ -330,12 +409,51 @@ def parse_map_file_armcc(self, file_desc):
             # Start decoding the map file
             for line in infile:
 
-                [name, size, section] = self.parse_section_armcc(line)
+                [object_name, object_size, section] = \
+                                self.parse_section_armcc(line)
 
-                if size == 0 or name == "" or section == "":
+                if object_size == 0 or object_name == "" or section == "":
                     pass
                 else:
-                    self.module_add(name, size, section)
+                    self.module_add(object_name, object_size, section)
+
+
+    def check_new_library_iar(self, line):
+        """
+        Searches for libraries and returns name. Example:
+        m7M_tls.a: [43]
+
+        """
+
+        rex_address_line = re.compile(r'^(.+\.a)\:.+$')
```

Comment:
This comment is the same as the other Regular expression related ones:
1) `re` not `rex` is common in python
2) Please make this a Class or global constant